charts/redpanda/templates/_statefulset.tpl

```diff
@@ -15,23 +15,24 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */}}
 
-{{- define "statefulset-pod-labels" -}}
+{{- define "statefulset-pod-labels-selector" -}}
 {{- /*
   StatefulSets cannot change their selector. Use the existing one even if it's broken.
   New installs will get better selectors.
 */ -}}
 {{- $sts := lookup "apps/v1" "StatefulSet" .Release.Namespace (include "redpanda.fullname" .) -}}
-{{- $labels := dig "spec" "selector" "matchLabels" "" $sts -}}
-{{- if not (empty $labels) -}}
-{{ $labels | toYaml }}
-{{- else -}}
-app.kubernetes.io/name: {{ template "redpanda.name" . }}
-app.kubernetes.io/instance: {{ .Release.Name | quote }}
-app.kubernetes.io/component: {{ (include "redpanda.name" .) | trunc 51 }}-statefulset
-{{- with .Values.commonLabels }}
-{{ toYaml . }}
-{{- end }}
+{{- $labels := dig "spec" "selector" "matchLabels" nil $sts -}}
+{{- get ((include "redpanda.StatefulSetPodLabelsSelector" (dict "a" (list . $labels))) | fromJson) "r" | toYaml }}
 {{- end -}}
+
+{{- define "statefulset-pod-labels" -}}
+{{- /*
+  StatefulSets cannot change their selector. Use the existing one even if it's broken.
+  New installs will get better selectors.
+*/ -}}
+{{- $sts := lookup "apps/v1" "StatefulSet" .Release.Namespace (include "redpanda.fullname" .) -}}
+{{- $labels := dig "spec" "template" "metadata" "labels" nil $sts -}}
```

Comment:
If it would be easier, I _think_ you could just pass in the entire sts and then handle it as a fully types StatefulSet in Go world. That might also solve the `nil` problem above?

Reply:
I can pass StatefulSet object. No problem.